tests/test_cmdline.py

```diff
@@ -946,6 +946,35 @@ def test_cmdline_args_in_combination_with_config_file(
     assert output.splitlines() == expect
 
 
+def test_deps_across_groups_appear_just_once_in_list_deps_detailed(tmp_path):
+    deps_data, uniq_deps = pyproject_toml_contents()
+    deps_path = tmp_path / "pyproject.toml"
+    exp_lines_from_pyproject = [f"{deps_path}: {dep}" for dep in uniq_deps]
+    deps_path.write_text(dedent(deps_data))
+    output, *_ = run_fawltydeps("--list-deps", "--detailed", f"--deps={deps_path}")
+    obs_lines = output.splitlines()
+    assert_unordered_equivalence(obs_lines, exp_lines_from_pyproject)
+
+
+def test_deps_across_groups_appear_just_once_in_order_in_general_detailed(tmp_path):
+    deps_data, uniq_deps = pyproject_toml_contents()
+    deps_path = tmp_path / "pyproject.toml"
+    deps_path.write_text(dedent(deps_data))
+    output, *_ = run_fawltydeps("--detailed", f"{tmp_path}")
+    obs_lines_absolute = output.splitlines()
+    obs_lines_relevant = dropwhile(
+        lambda line: not line.startswith(UNUSED_DEPS_OUTPUT_PREFIX), obs_lines_absolute
+    )
+    next(obs_lines_relevant)  # discard
+    unused_deps = [UnusedDependency(name, [Location(deps_path)]) for name in uniq_deps]
+    exp_lines = [
+        line
+        for dep in unused_deps
+        for line in f"- {dep.render(include_references=True)}".split("\n")
+    ]
+    assert list(obs_lines_relevant) == exp_lines
+
+
```

Comment:
NIt: Why move these tests to the other side of `pyproject_toml_contents()`? Personally, I have a slight preference for defining the callee (`pyproject_toml_contents()`) above the callers (the tests).

Reply:
No _strong_ preference, but in general I like the structure...
1. fixtures
2. tests
3. helpers

(disregarding imports / constants at the top)